st2auth_ldap_backend/ldap_backend.py

```diff
@@ -160,15 +164,52 @@ def _ldap_search(self, connection, username, criteria):
                          (base_dn, search_filter, scope, pattern)
         :type criteria: ``dict``
         """
+        results = []
         base_dn = criteria['base_dn']
         search_filter = criteria['search_filter'].format(username=username)
         scope = self._scope_to_ldap_option(criteria['scope'])
 
         LOG.debug('Searching ... %s %s %s' % (base_dn, scope, search_filter))
-        result = connection.search_s(base_dn, scope, search_filter)
+
+        result_id = connection.search(base_dn, scope, search_filter)
+        while 1:
```

Comment:
Can we make this adhere to a timeout? I don't like endless loops so if we could do a 10, 20 second timeout I'd feel much better.

Also it's unclear to me why we need a loop here. Is there a more deterministic way to get the results without iterating over the connection? This last part is my own ignorance to the subject so if you could clarify that'd be great.